mesonbuild/backend/ninjabackend.py

```diff
@@ -147,7 +147,7 @@ def ninja_quote(text: str, is_build_line=False) -> str:
     return quote_re.sub(r'$\g<0>', text)
 
 class TargetDependencyScannerInfo:
-    def __init__(self, private_dir: str, source2object: T.Dict[str, str]):
+    def __init__(self, private_dir: str, source2object: T.Dict[str, str]) -> None:
```

Comment:
Oh neat, I did not realize that.

Reply:
> Adding -> None: to __init__ is only requierd when there are no arguments other than self.

TIL. Is there a reason why it's not working when there is only self? That seems very unexpected. TBH I never understood why `->None` is required at all anywhere.

Reply:
> Is there a reason why it's not working when there is only self? That seems very unexpected.

It's so that code that's in in the process of being typed doesn't generate a billion errors for untyped attributes in their initializers.

> TBH I never understood why ->None is required at all anywhere.

Because the default is that mypy assumes `-> Any`.

Reply:
I forgot one of these, hold on...